src/app/vault/components/VaultUserMetricsContainer.tsx

```diff
@@ -0,0 +1,56 @@
+'use client'
+
+import { BalanceInfo } from '@/components/BalanceInfo'
+import { useVaultBalance } from '../hooks/useVaultBalance'
+import { useGetAddressBalances } from '@/app/user/Balances/hooks/useGetAddressBalances'
+import { useAccount } from 'wagmi'
+import { formatSymbol } from '@/app/shared/formatter'
+import { USDRIF } from '@/lib/constants'
+import { VaultActions } from './VaultActions'
+import { CircularProgress } from './CircularProgress'
+
+/**
+ * Component displaying user-specific vault metrics and actions
+ * Shows wallet balance, total supplied, and supply/withdraw buttons
+ */
+export const VaultUserMetricsContainer = () => {
+  const { totalAssets, userBalance, isLoading } = useVaultBalance()
+  const { balances, isBalancesLoading } = useGetAddressBalances()
+  const { isConnected } = useAccount()
+  const userUsdrifBalance = balances[USDRIF]
+
+  // Calculate percentage of vault owned by user
+  const vaultPercentage = totalAssets > 0 ? (Number(userBalance) / Number(totalAssets)) * 100 : 0
+
+  return (
+    <div className="flex flex-col gap-6 w-full">
+      <div className="flex flex-row flex-wrap gap-6 items-center w-full">
+        {isConnected && (
+          <BalanceInfo
+            className="max-w-[214px] min-w-[180px]"
+            title="Your wallet balance"
+            amount={isBalancesLoading ? '...' : userUsdrifBalance.formattedBalance}
+            symbol="USDRIF"
+          />
+        )}
+        <BalanceInfo
+          className="max-w-[214px] min-w-[180px]"
+          title="Total Deposited"
+          amount={isLoading ? '...' : formatSymbol(userBalance, 'USDRIF')}
```

Comment:
should we use 0 decimals, if I deposit $100 for me to generate $1 at a 5% APY Ill need to wait almost 3 months.